### PR TITLE
Fix for LmToProjData (issue #61)-check if compatible template

### PR DESCRIPTION
**Pull request by eliseemond**
see #61 

Jannis (@jafische), can you check if the fix in src/include/stir/listmode/CListRecordSAFIR.h is correct?
The new method "is_valid_template", used within lm_to_projdata to check whether the input template is compatible, returns always true in the case of CListEventSAFIR as the crash should never occur.
Are you happy with this modification?

Cheers,
Elise

### CODE REVIEW

**Comment by jafische** [comment]
Hi Elise,
let me rephrase to make sure I understand it correctly: the function should return true, if the class delivers non-arc-corrected data to STIR? This is indeed true for CListRecordSAFIR as it returns the LORAs2Points with the two points being the physical crystal coordinates as specified in the crystal map.
So yes, I'm happy :).
Cheers,
Jannis

**Comment by KrisThielemans** [comment]
hi, no, the function should return true if the class can support the template (as Elise I think nicely wrote in the doxygen). In most cases, the CList code can only handle non-arccorrected data (and therefore crashed when given an arccorrected template), but we're not sure if SAFIR would be able to support arccorrected data directly.

**Comment by jafische** [comment]
Hi Elisa, hi Kris,

I would say the SAFIR code can support both because it's up to the template projection data ("virtual scanner") object to decide, in which bin an event will be placed. The SAFIR class used the default implementation of  stir::CListEvent::get_bin(Bin&, const ProjDataInfo &) and only implements LORAs2Points<float> stir::CListEvent::get_LOR(). This does not yet depend on whether the data is arc-corrected or not. So I guess, it depends on how stir::CListEvent::get_bin(...) behaves. @KrisThielemans: What do you think?

Cheers

Jannis

**Comment by KrisThielemans** [comment]
sounds ok to me, so I'll accept this.
